### ISSUE TITLE
Codec specific configuration parameters

### ISSUE THREAD

**Issue by chcunningham** [WICG/web-codecs#97]
Tracking spec work to allow for codec specific tuning of encoder/decoder configuration. First example of this will be AVC vs Annex-B for H264. 

Probably something like:

```
configure({
  contentType: 'avc....';
  avcOptions: {
     format: "annex-b"
  }
});
```

**Comment by code-bythepound** [comment]
Just to clarify here, is the thought to make the codec string match the current string for H264 ie: "avc1.profile.compat.level", but add a new contentType and avcOptions?  What should go in the "description" (extradata) field in this case?
Would the annex-b format stream then allow sps/pps in-band rather than require it out of band?
Thanks.

**Comment by sandersdan** [comment]
I don't think the intent was to propose changing from `codec` to `contentType`, that's a small detail that can be bikeshed separately.

Rather, the bitstream `format` and AVC `description` are the first examples we have of codec-specific parameters, and the proposal is to move them into a codec-specific grouping.

**Comment by code-bythepound** [comment]
Got it thanks.  So this isn't about adding annex-b format specifically, it is about changing the dictionary passed to configure.  Is there a separate bug for adding annex-b format for h264 streams?

**Comment by sandersdan** [comment]
#72

**Comment by chcunningham** [comment]
Fixed by #117. More configuration options and more codecs will surely follow, but we'll open separate issues for those as needed.
